src/main/java/com/stefancooper/SpigotUHC/types/AdditionalEnchants.java

```diff
@@ -0,0 +1,125 @@
+package com.stefancooper.SpigotUHC.types;
+
+import com.stefancooper.SpigotUHC.Config;
+import com.stefancooper.SpigotUHC.Defaults;
+import com.stefancooper.SpigotUHC.enums.ConfigKey;
+import com.stefancooper.SpigotUHC.utils.Utils;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import static com.stefancooper.SpigotUHC.utils.Utils.romanNumeral;
+
+public class AdditionalEnchants {
+
+    private final Config config;
+    private static final String NIGHT_VISION_LORE = ChatColor.GRAY + "" + ChatColor.ITALIC + "Night Vision Goggles";
+
+    public AdditionalEnchants(Config config) {
+        this.config = config;
+    }
+
+    public void apply(ItemStack item) {
+        if (item == null || item.getType() == Material.AIR) return;
+
+        Material type = item.getType();
+
+        switch (type) {
+            case SHIELD -> {
+                if (config.getProperty(ConfigKey.ADDITIONAL_ENCHANTS_SHIELD, Defaults.ADDITIONAL_ENCHANTS_SHIELD)) {
+                    // Block re-enchanting if already enchanted
+                    if (!item.getEnchantments().isEmpty()) return;
+
+                    ItemMeta meta = item.getItemMeta();
+                    if (meta == null) return;
+
+                    List<String> lore = meta.hasLore() ? new ArrayList<>(meta.getLore()) : new ArrayList<>();
+
+                    boolean hasCustomShieldEnchant = lore.stream().anyMatch(line ->
+                            ChatColor.stripColor(line).toLowerCase().startsWith("knockback") ||
+                                    ChatColor.stripColor(line).toLowerCase().startsWith("thorns"));
+
+                    if (hasCustomShieldEnchant) return;
+
+                    boolean applyKnockback = new Random().nextBoolean();
+
+                    if (applyKnockback) {
+                        int knockbackLevel = 1 + new Random().nextInt(2); // Knockback I–II
+                        item.addUnsafeEnchantment(Enchantment.KNOCKBACK, knockbackLevel);
+                        lore.add(ChatColor.GRAY + "Knockback " + romanNumeral(knockbackLevel));
+                    } else {
+                        int thornsLevel = 1 + new Random().nextInt(3); // Thorns I–III
+                        item.addUnsafeEnchantment(Enchantment.THORNS, thornsLevel);
+                        lore.add(ChatColor.GRAY + "Thorns " + romanNumeral(thornsLevel));
+                    }
```

Comment:
i am confused by this, isn't all of this randomness done in the events class? what is `AdditonalEnchants` for?

Reply:
lets call cos i'm probably missing something obvious